openmc/tallies.py

```diff
@@ -1482,19 +1482,30 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Create copies of self and other tallies to rearrange for tally
+        # arithmetic
+        self_copy = copy.deepcopy(self)
+        other_copy = copy.deepcopy(other)
```

Comment:
There seems to be an issue in future operations if the `self` and `other` filters are swapped inline, so I had to make copies of each tally involved in the arithmetic. I believe this has to do with the way the transport cross section is computed in `mgxs.py`. Specifically, in `TransportXS.compute_xs()`, the `energyout` filter is assumed to be the last filter (i.e. indexed by -1). If the filters can be swapped inline, this is not guaranteed to be true so hence the need for the copy here. I'm open to suggestions on how to get around this copy since it could lead to performance bottlenecks.